deployments/extensions.go

```diff
@@ -466,6 +538,134 @@ func (e *Extension) Uninstall(c *kubernetes.Cluster, ui *ui.UI, options *kuberne
 	return nil
 }
 
+// get extension from the registry
+func (e *Extension) getExtension(ui *ui.UI, fusemlURL string) error {
+
+	fullURL := fmt.Sprintf("%s/extensions/%s", fusemlURL, e.Desc.Product)
+
+	resp, err := http.Get(fullURL)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == 200 {
+
+		var buf bytes.Buffer
+		_, _ = io.Copy(&buf, resp.Body)
+
+		fmt.Printf("response body as string: \n%s\n", buf.String())
+
+		ext := registeredExtension{}
+
+		err = json.Unmarshal(buf.Bytes(), &ext)
+		if err != nil {
+			return errors.Wrap(err, "failed to parse description file")
+		}
+		ui.Exclamation().Msg(fmt.Sprintf("extension already present with id %s, product %s", *ext.ID, *ext.Product))
+	} else if resp.StatusCode == 404 {
+		ui.Exclamation().Msg(fmt.Sprintf("extension %s not yet registered, processing now", e.Name))
+	} else {
+		return errors.New(fmt.Sprintf("Unexepcted response from registry: %d", resp.StatusCode))
+	}
+	return nil
+}
+
+// Unregister extension from the extension registry
+func (e *Extension) UnRegister(c *kubernetes.Cluster, ui *ui.UI, options *kubernetes.InstallationOptions) error {
+	domain, err := options.GetString("system_domain", "")
+	if err != nil {
+		return errors.New("system_domain value not provided")
+	}
+
+	fusemlURL := fmt.Sprintf("http://%s.%s", CoreDeploymentID, domain)
+	fullURL := fmt.Sprintf("%s/extensions/%s", fusemlURL, e.Desc.Product)
+
+	// no direct Delete method so we have to create a client and a request
+	client := &http.Client{}
+
+	req, err := http.NewRequest("DELETE", fullURL, nil)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == 204 {
+		return nil
+	}
+
+	// no such extension found, that might be OK...
+	if resp.StatusCode == 404 {
+		return nil
+	}
+
+	fmt.Println("response from DELETE: ", resp.StatusCode)
+
+	// something else is wrong, read the response from POST call
+	var buf bytes.Buffer
+	_, _ = io.Copy(&buf, resp.Body)
+
+	return errors.New(fmt.Sprintf("Failed unregistering the extension. Server returns %s: ", buf.String()))
+}
+
+// Register extension in the registry that is run by fuseml-core server
+func (e *Extension) Register(c *kubernetes.Cluster, ui *ui.UI, options *kubernetes.InstallationOptions) error {
+
+	domain, err := options.GetString("system_domain", "")
+	if err != nil {
+		return errors.New("system_domain value not provided")
+	}
+
+	fusemlURL := fmt.Sprintf("http://%s.%s", CoreDeploymentID, domain)
+	fullURL := fmt.Sprintf("%s/extensions", fusemlURL)
+
+	extServices := []*registeredExtensionService{}
+	for _, service := range e.Desc.Services {
+		extSetvice := registeredExtensionService{
+			ID:          service.ID,
+			ExtensionID: &e.Name,
```

Comment:
hint: you don't need to set the extension ID for services during registration, the core service can deduce it automatically, because you're registering everything together as a hierarchy of objects

Reply:
OK